packages/core/content-type-builder/admin/src/components/ReviewWorkflowsToggle/index.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Checkbox } from '@strapi/design-system';
+import { ConfirmDialog } from '@strapi/helper-plugin';
+import { getTrad } from '../../utils';
+
+const ReviewWorkflowsToggle = ({
+  description,
+  disabled,
+  intlLabel,
+  isCreating,
+  name,
+  onChange,
+  value,
+}) => {
+  const { formatMessage } = useIntl();
+  const [showWarning, setShowWarning] = useState(false);
+  const label = intlLabel.id
+    ? formatMessage(
+        { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
+        { ...intlLabel.values }
+      )
+    : name;
+
+  const hint = description
+    ? formatMessage(
+        { id: description.id, defaultMessage: description.defaultMessage },
+        { ...description.values }
+      )
+    : '';
+
+  const handleToggle = () => setShowWarning((prev) => !prev);
+
+  const handleConfirm = () => {
+    onChange({ target: { name, value: false } });
+
+    handleToggle();
+  };
+
+  const handleChange = ({ target: { checked } }) => {
+    if (!checked && !isCreating) {
+      handleToggle();
+
+      return;
+    }
+
+    onChange({ target: { name, value: checked } });
+  };
+
+  return (
+    <>
+      <Checkbox checked={value} disabled={disabled} hint={hint} name={name} onChange={handleChange}>
+        {label}
+      </Checkbox>
+
+      <ConfirmDialog
+        isOpen={showWarning}
+        onToggleDialog={handleToggle}
+        onConfirm={handleConfirm}
+        bodyText={{
+          id: getTrad('popUpWarning.review-workflows.message'),
+          defaultMessage:
+            'If you disable review workflow, the associated workflow-data with this content-type will be deleted.',
```

Comment:
@maevalienard I think we haven't defined the confirmation message yet. Do you have input here?

Reply:
This is what I wrote in Figma: "If you disable the review workflows feature, your stages will be deleted. Are you sure you want to disable it?" - This is pretty similar to what we have for D/P and i18n. Does it seem clear enough to you?

Reply:
The thing is it is not entirely true because the stages will not be deleted. The association between a stage/ workflow and the entities belonging to a content-type will be deleted (the relation if you will).

Reply:
True. How about: "If you disable the review workflows feature, your stages will be removed from the Content Manager. Are you sure you want to disable it?" ?

Reply:
Counter proposal :) 

"If you disable the review workflows feature, all stage-related information will be removed for this content-type. Are you sure you want to disable it?"

Reply:
Perfect 👌